linalg/densemat.hpp

```diff
@@ -150,11 +150,15 @@ class DenseMatrix : public Matrix
    /// Multiply a vector with the transpose matrix.
    virtual void MultTranspose(const Vector &x, Vector &y) const;
 
+   using Operator::Mult;
+   using Operator::MultTranspose;
+
    /// y += A.x
-   void AddMult(const Vector &x, Vector &y) const;
+   virtual void AddMult(const Vector &x, Vector &y, const double a = 1.0) const;
 
    /// y += A^t x
-   void AddMultTranspose(const Vector &x, Vector &y) const;
+   virtual void AddMultTranspose(const Vector &x, Vector &y,
+                                 const double a = 1.0) const;
 
    /// y += a * A.x
    void AddMult_a(double a, const Vector &x, Vector &y) const;
```

Comment:
What's now the difference between `AddMult_a` and `AddMult`? From a user-perspective this looks like the same method just with differently ordered arguments.

Reply:
That's correct. I figured keeping both allows the `a = 1.0` case to still be optimized compared to the general case at the expense of a single `if` check. It also keeps consistency with the overloads for `DenseMatrix` arguments, and means no changes to other code calling `AddMult_a` or `AddMult`.

If you think they should be unified that is fine with me and I will change it.